src/accounts/tezos.ts

```diff
@@ -0,0 +1,99 @@
+import { Account } from "./account";
+import { BaseMessage, Chain } from "../messages/message";
+import { GetVerificationBuffer } from "../messages";
+import { InMemorySigner } from "@taquito/signer";
+import { b58cdecode, b58cencode, prefix } from "@taquito/utils";
+import { getPkhfromPk } from "@taquito/utils";
+
+import nacl from "tweetnacl";
+
+/**
+ * XTZAccount implements the Account class for the Tezos protocol.
+ * It is used to represent a Tezos account when publishing a message on the Aleph network.
+ */
+export class TEZOSAccount extends Account {
+    private signer: InMemorySigner;
+
+    /**
+     * @param publicKey The public key encoded in base58. Needed due to asynchronous getter of the public key.
+     * @param signer The signer containing the private key used to sign the message.
+     */
+    constructor(publicKey: string, signer: InMemorySigner) {
```

Comment:
The account class does not need the `publicKey` attribute anymore, but the address is still mandatory, if those two parameters are the same (publicKey === address), can you rename the parameter in the constructor so it's less confusing ?

Reply:
Well, the problem here is that `InMemorySinger.GetPublicKey()` is async, therefore I cannot invoke it inside the constructor.